### ISSUE TITLE
`dokku apps:clone` can't work on CI because it always exits 1.

### ISSUE THREAD

**Issue by tamanobi** [dokku/dokku#3418]
## Description of problem
When I try to do the dokku tutorial(http://dokku.viewdocs.io/dokku/community/tutorials/deploying-with-gitlab-ci/#review-applications), I noticed that exit status of `dokku apps:clone` will be 1. But cloned app will work well.

I got Permission Denied because find walks cache (group id: 32767) directory.

I focus just exit status.

### How reproducible
Everytime.

### Steps to Reproduce

1. dokku apps:clone old-app new-app

#### Actual Results
```
$ dokku apps:clone like-dler like-dler2
-----> Creating like-dler2... done
find: ‘./cache/.heroku/python/lib’: Permission denied
find: ‘./cache/.heroku/python/include’: Permission denied
$ echo $?
1
```

#### Expected Results
`dokku apps:clone like-dler like-dler2` returns 0

## Environment Information
- dokku version: 0.14.2
- OS: 18.04.1 LTS (Bionic Beaver)
- Shell: GNU bash, version 4.4.19(1)-release (x86_64-pc-linux-gnu)
- find version: find (GNU findutils) 4.7.0-git

### `dokku report APP_NAME` output
Full log is here: https://gist.github.com/tamanobi/582499375dd9a319d83afd8dace2be51

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?:
VPS(Sakura VPS)

### Additional information
I created patch to solve this issue and patch my `dokku`. The patch is here: https://github.com/tamanobi/dokku/pull/1/files
After this discussion, I will make a PR.

P.S. I created patch(https://github.com/dokku/dokku/pull/3434).

**Comment by josegonzalez** [comment]
Can you show the output of:

```
dokku --trace apps:clone like-dler like-dler3
```

**Comment by tamanobi** [comment]
@josegonzalez Thanks for your reply. The log is shown below.

```
$ dokku --trace apps:clone like-dler like-dler3
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOKKU_IMAGE=gliderlabs/herokuish
+ DOKKU_IMAGE=gliderlabs/herokuish
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export DOKKU_GLOBAL_RUN_ARGS=--label=dokku
+ DOKKU_GLOBAL_RUN_ARGS=--label=dokku
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args apps:clone like-dler like-dler3
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=2
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=3
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=4
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ apps:clone =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ /usr/bin/tty
+ [[ /dev/pts/7 == \n\o\t\ \a\ \t\t\y ]]
+ return 0
++ id -un
+ [[ dokku != \d\o\k\k\u ]]
+ [[ apps:clone =~ ^plugin:.* ]]
+ [[ apps:clone == \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ -n '' ]]
+ dokku_auth apps:clone like-dler like-dler3
+ declare 'desc=calls user-auth plugin trigger'
+ export SSH_USER=rin
+ SSH_USER=rin
+ export SSH_NAME=default
+ SSH_NAME=default
+ plugn trigger user-auth rin default apps:clone like-dler like-dler3
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ [[ ! -n '' ]]
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ [[ apps:clone =~ (ssh-keys:add|ssh-keys:remove) ]]
+ return 0
+ case "$1" in
+ execute_dokku_cmd apps:clone like-dler like-dler3
+ declare 'desc=executes dokku sub-commands'
+ local PLUGIN_NAME=apps:clone
+ local PLUGIN_CMD=apps:clone
+ local implemented=0
+ local script
+ argv=("$@")
+ local argv
+ case "$PLUGIN_NAME" in
++ readlink -f /var/lib/dokku/plugins/enabled/apps
+ [[ /var/lib/dokku/core-plugins/available/apps == *core-plugins* ]]
+ [[ apps:clone == \a\p\p\s\:\c\l\o\n\e ]]
+ shift 1
+ [[ ! -z '' ]]
+ set -- apps:clone like-dler like-dler3
+ [[ -x /var/lib/dokku/plugins/enabled/apps:clone/subcommands/default ]]
+ [[ -x /var/lib/dokku/plugins/enabled/apps:clone/subcommands/apps:clone ]]
+ [[ -x /var/lib/dokku/plugins/enabled/apps/subcommands/clone ]]
+ [[ -n clone ]]
+ /var/lib/dokku/plugins/enabled/apps/subcommands/clone apps:clone like-dler like-dler3
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/plugins/available/apps/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
+ source /var/lib/dokku/plugins/available/ps/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
+ apps_clone_cmd apps:clone like-dler like-dler3
+ declare 'desc=clones an app'
+ argv=("$@")
+ declare cmd=apps:clone argv
+ [[ apps:clone == \a\p\p\s\:\c\l\o\n\e ]]
+ shift 1
+ declare OLD_APP=like-dler NEW_APP=like-dler3
+ local IGNORE_EXISTING=false
+ local SKIP_REBUILD=false
+ for arg in "$@"
+ [[ like-dler == \-\-\s\k\i\p\-\d\e\p\l\o\y ]]
+ [[ like-dler == \-\-\i\g\n\o\r\e\-\e\x\i\s\t\i\n\g ]]
+ [[ like-dler =~ ^--.* ]]
+ break
+ OLD_APP=like-dler
+ NEW_APP=like-dler3
+ [[ -z like-dler ]]
+ is_valid_app_name like-dler
+ declare 'desc=verify app name format'
+ local APP=like-dler
+ [[ ! -n like-dler ]]
+ [[ like-dler =~ ^[a-z].* ]]
+ [[ ! like-dler =~ [A-Z] ]]
+ [[ ! like-dler =~ [:] ]]
+ return 0
+ is_valid_app_name like-dler3
+ declare 'desc=verify app name format'
+ local APP=like-dler3
+ [[ ! -n like-dler3 ]]
+ [[ like-dler3 =~ ^[a-z].* ]]
+ [[ ! like-dler3 =~ [A-Z] ]]
+ [[ ! like-dler3 =~ [:] ]]
+ return 0
+ apps_exists like-dler
+ apps_exists like-dler3
+ local NEW_CACHE_DIR=/home/dokku/like-dler3/cache
+ local NEW_CACHE_HOST_DIR=/home/dokku/like-dler3/cache
+ apps_create like-dler3
+ declare 'desc=verifies app name and creates an app'
+ declare APP=like-dler3
+ is_valid_app_name like-dler3
+ declare 'desc=verify app name format'
+ local APP=like-dler3
+ [[ ! -n like-dler3 ]]
+ [[ like-dler3 =~ ^[a-z].* ]]
+ [[ ! like-dler3 =~ [A-Z] ]]
+ [[ ! like-dler3 =~ [:] ]]
+ return 0
+ apps_exists like-dler3
+ mkdir -p /home/dokku/like-dler3
+ dokku_log_info1_quiet 'Creating like-dler3... done'
+ declare 'desc=log info1 formatter (with quiet option)'
+ [[ -z '' ]]
+ echo '-----> Creating like-dler3... done'
-----> Creating like-dler3... done
+ plugn trigger post-create like-dler3
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ [[ ! -n '' ]]
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/plugins/available/docker-options/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ AVAILABLE_PHASES=(build deploy run)
+ source /var/lib/dokku/plugins/available/ps/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
+ ps_post_create like-dler3
+ declare 'desc=ps post-create plugin trigger'
+ local trigger=ps_post_create
+ local APP=like-dler3
+ passed_phases=(deploy)
+ local passed_phases
+ add_passed_docker_option 'passed_phases[@]' --restart=on-failure:10
+ declare 'desc=adds docker option to specified phases'
+ passed_phases=("${!1}")
+ local passed_phases
+ shift
+ local passed_option_string=--restart=on-failure:10
+ local phase
+ for phase in "${passed_phases[@]}"
++ get_phase_file_path deploy
++ declare 'desc=return docker-options config file path for specified phase'
++ local phase_file_prefix=DOCKER_OPTIONS_
++ local phase=deploy
++ [[ -n like-dler3 ]]
++ [[ -n deploy ]]
++ fn-get-phase-file-path like-dler3 deploy
++ declare 'desc=return docker-options config file path for specified phase'
++ local APP=like-dler3 PHASE=deploy
++ local PHASE_FILE=/home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY
++ echo /home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY
+ local phase_file_path=/home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY
+ create_phase_file_if_required /home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY
+ declare 'desc=create phase file'
+ local phase_file_path=/home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY
+ [[ -f /home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY ]]
+ touch /home/dokku/like-dler3/DOCKER_OPTIONS_DEPLOY
+ echo --restart=on-failure:10
+ local all_phase_options=--restart=on-failure:10
+ echo -e --restart=on-failure:10
+ sed '/^$/d'
+ sort -u
+ scheduler-docker-local-post-create like-dler3
+ declare 'desc=scheduler-docker-local post-create plugin trigger'
+ declare trigger=scheduler-docker-local-post-create
+ declare APP=like-dler3
+ mkdir -p /var/lib/dokku/data/scheduler-docker-local/like-dler3
+ chown -R dokku:dokku /var/lib/dokku/data/scheduler-docker-local/like-dler3
+ pushd /home/dokku/like-dler/.
+ grep -v ./cache
+ find ./CONTAINER.worker.1 ./DOCKER_OPTIONS_BUILD ./DOCKER_OPTIONS_DEPLOY ./DOCKER_OPTIONS_RUN ./DOKKU_SCALE ./ENV ./HEAD ./branches ./cache ./config ./description ./hooks ./info ./objects ./refs -not '(' -name .cache ')'
+ cpio -pdmu --quiet /home/dokku/like-dler3
find: ‘./cache/.heroku/python/lib’: Permission denied
find: ‘./cache/.heroku/python/include’: Permission denied
```

**Comment by josegonzalez** [comment]
Whats interesting there is that the grep appears to be applied _before_ the find call...

**Comment by josegonzalez** [comment]
Okay I think this is a result of Ubuntu 18.04 being a bit stricter around file permissions. I just tested your fix on both Ubuntu 16.04 and 18.04, and 18.04 has the issue, while 16.04 does not.
